parsing/parser.mly

```diff
@@ -2130,6 +2105,30 @@ let_pattern:
       { $1 }
 ;
 
+%inline array_set(dot, left, mid, right):
+  | simple_expr dot left mid right LESSMINUS expr
+      { $1, $2, $3, $4, Some $7 }
+;
+%inline array_get(dot, left, mid, right):
+  | simple_expr dot left mid right
+      { $1, $2, $3, $4, None }
+  | simple_expr dot left mid error
+      { let left, right = paren_to_strings $3 in
+        unclosed left $loc($3) right $loc($5)
+      }
+;
+%inline all_array_expr(op, dot, mid):
+  | op(dot, LPAREN { Paren }, mid, RPAREN)       { $1 }
```

Comment:
Another way to approach this could be

```
%inline parens(mid): | LPAREN v=mid RPAREN { (Paren, v) }

%inline all_array_expr(op, dot, mid)
| op(dot, parens(mid)) { $1 }
```